internal/compiler-interface/src/main/java/xsbti/compile/JavaTool.java

```diff
@@ -21,9 +21,11 @@
    *
    * @param sources  The list of java source files to compile.
    * @param options  The set of options to pass to the java compiler (includes the classpath).
+   * @param classFileManager The manager for generated class files.
    * @param reporter The reporter for semantic error messages.
    * @param log      The logger to dump output into.
    * @return true if no errors, false otherwise.
    */
-  boolean run(File[] sources, String[] options, Reporter reporter, Logger log);
+  boolean run(File[] sources, String[] options, ExternalHooks.ClassFileManager classFileManager,
```

Comment:
It is necessary to add `classFileManager` as a parameter here? If the majority of the time it's passed null and if it doesn't change that much, I feel like it belongs more to the setup of the surrounding class, unless I'm not understanding something.